### PR TITLE
Release - 1.1.0

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Features:
- https://github.com/nimblehq/nimble-survey-web/pull/25 As an API user, I can sign up and login with the correct applications
- https://github.com/nimblehq/nimble-survey-web/pull/20 [Frontend] As a user, I can see the reset password screen

Chores:
- https://github.com/nimblehq/nimble-survey-web/pull/31 Add BASIC_AUTHENTICATION ARGs to build docker
- https://github.com/nimblehq/nimble-survey-web/pull/32 Revert removing sprockets to support the external gems
- https://github.com/nimblehq/nimble-survey-web/pull/33 Add workflow_dispatch trigger for deploy workflow
- https://github.com/nimblehq/nimble-survey-web/pull/34 Enable force_ssl on production
- https://github.com/nimblehq/nimble-survey-web/pull/38 Allow CORS request
 
## Insight
https://github.com/nimblehq/nimble-survey-web/milestone/2
 

## Proof Of Work
[Staging](https://nimble-survey-web-staging.herokuapp.com/oauth/applications)
[Postman PR](https://nimblehq.postman.co/collections/11835486-63105f38-c20f-493d-b4d8-873a18c8d8c7/pull-requests/0757167e-6a50-495c-8765-75520c464e83?view=overview&workspace=9daf6b25-882e-4ad0-8299-95529e2883ff)

### CODE REVIEW

File: app/controllers/api/v1/application_controller.rb [hunk 1]
```diff
@@ -11,19 +13,15 @@ def current_user
...
-        errors = [
-          {
-            source: source,
-            detail: detail,
-            code: code,
-            meta: meta
-          }.compact
-        ]
+      # :reek:FeatureEnvy
+      def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/api/v1/application_controller.rb:17]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/app/controllers/api/v1/application_controller.rb#L17"></span>Remove unused methods (api::v1::applicationcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/tokens_controller.rb [hunk 2]
```diff
@@ -3,23 +3,15 @@
...
-          revoke_token
-          render json: {}
-        else
-          render json: revocation_error_response, status: :forbidden
-        end
+      include API::V1::ErrorHandler
+
+      private
+
+      def revocation_error_response
```
Comment (team-nimblehq) [inline app/controllers/api/v1/tokens_controller.rb:10]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/app/controllers/api/v1/tokens_controller.rb#L10"></span>Remove unused methods (api::v1::tokenscontroller#revocation_error_response)</td>
    </tr>
  </tbody>
</table>

File: config/routes.rb [hunk 3]
```diff
@@ -1,14 +1,23 @@
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users, skip: %i[sessions passwords registrations]
   devise_scope :user do
-    root to: "devise/registrations#edit"
+    resource :password, only: %i[edit update], as: 'user_password'
```
Comment (team-nimblehq) [inline config/routes.rb:4]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/config/routes.rb#L4"></span>Restrict auto-generated routes password (only: [:update])</td>
    </tr>
  </tbody>
</table>

File: spec/support/authentication_helpers.rb [hunk 4]
```diff
@@ -4,9 +4,17 @@
...
+    application = Fabricate(:application)
+    access_token = Fabricate(:access_token, resource_owner_id: user.id, application_id: application.id)
 
     allow(controller).to receive(:doorkeeper_token) { access_token }
     request.headers['Content-Type'] = 'application/json'
   end
+
+  def oauth_application_params(application = Fabricate(:application))
+    {
+      client_id: application.uid,
```
Comment (team-nimblehq) [inline spec/support/authentication_helpers.rb:16]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/authentication_helpers.rb#L16"></span>Refers to ‘application’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: spec/support/authentication_helpers.rb [hunk 5]
```diff
@@ -4,9 +4,17 @@
...
+      client_secret: application.secret
```
Comment (team-nimblehq) [inline spec/support/authentication_helpers.rb:17]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/authentication_helpers.rb#L17"></span>Refers to ‘application’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/app/controllers/api/v1/application_controller.rb#L12">app/controllers/api/v1/application_controller.rb#L12</a> - Remove unused methods (api::v1::applicationcontroller#current_user)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/disable_animation.rb#L8">spec/support/disable_animation.rb#L8</a> - Assumes too much for instance variable ‘@body’</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/disable_animation.rb#L8">spec/support/disable_animation.rb#L8</a> - Assumes too much for instance variable ‘@headers’</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/disable_animation.rb#L8">spec/support/disable_animation.rb#L8</a> - Assumes too much for instance variable ‘@status’</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/disable_animation.rb#L13">spec/support/disable_animation.rb#L13</a> - Has approx 7 statements</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/10e383cbcb61a960ca8de7fb15a118e04aaac39f/spec/support/disable_animation.rb#L20">spec/support/disable_animation.rb#L20</a> - Manually dispatches method call</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by olivierobert — APPROVED** [review]

**Comment by olivierobert** [comment]
As a codebase improvement, it would be better to remove Danger checks for the `spec/support` directory 💭 

image

**Comment by hoangmirs** [comment]
@olivierobert 
This issue comes from reek: https://github.com/troessner/reek/issues/1196
If we run reek file by file via Danger, it will ignore the `exclude_paths` setting and still run the check
I will try fixing it and create a PR to Danger reek

**Pull request merged**